syntax = "proto3";

package vegvisir.protocol.datatype;

import "vegvisirCommon.proto";

message HashSet {
    repeated bytes hashes = 1;
}

message Request {

    enum Type {
        DUMMY_REQUEST = 0;
        SEND_FRONTIER_SET = 1;
        SEND_BLOCK = 2;
        ADD_BLOCK = 3;
        LOCAL_RECONCILIATION_NEEDED = 4;
        REMOTE_RECONCILIATION_NEEDED = 5;
    }

    Type type = 1;
   
    oneof targets_type {
        HashSet targetHash = 2;
        vegvisir.common.datatype.AddBlocks add = 3;
    }
} 

message Response {
    oneof response_types {
        HashSet hashResponse = 1;
        vegvisir.common.datatype.AddBlocks blockResponse = 2;
    }
}

message FrontierMessage {
    oneof frontier_message_type {
        Request request = 1;
        Response response = 2;
    }
}
